src/lavinmq/federation/link.cr

```diff
@@ -323,6 +323,8 @@ module LavinMQ
           ::AMQP::Client.start(upstream_uri) do |c|
             ch = c.channel
             ch.queue_delete(@upstream_q)
+            ch.exchange_delete(@upstream_q)
+            ch.exchange_delete(@upstream_exchange)
```

Comment:
I'm not sure that we can delete this exchange unconditionally. It's not sure that we're creating it. We should probably not delete it no matter what...

Reply:
Maybe not the regular one, i added that as a guess, but the internal exchange should be deleted at cleanup